### DISCUSSION TITLE
Restart container without triggering a new build

### DISCUSSION THREAD

**Discussion by thereis** [coollabsio/coolify#3560] [Improvement Requests]
It would be cool if we could have a feature where we can simply restart the container in the server rather than triggering a new build of it.

**Comment by Nexulo** [comment]
This function would be extremely useful. Just because you change an Env Var, you don't necessarily have to make a new deployment. My docker compose stack is very extensive and it's really time-consuming to build. If you need 10-15 minutes for a new build every time just because you want to transfer a changed env var to the Docker container.

**Reply by djsisson** [reply]
a new env requires a new container, not just a restart due to container hash, so if your container is based off of a build and not an image, then a new build has to happen.
The only option is to build seperate and just reference the image, that way a restart would use new env but old image saving the rebuild.

**Comment by andrasbacsai** [comment]
So this is only for docker-compose based applications & services, right?

**Reply by thereis** [reply]
Yup! I do have services that have a custom Dockerfile containers, I just want to restart them with the new env vars rather than rebuilding.

**Reply by Nexulo** [reply]
I think it would be extremely useful to have the following buttons for a docker compose application/service:
- stop
- start
- restart

**Reply by chris-si** [reply]
It should also be implemented for git-based deployments which e.g. use a Dockerfile as described in #2745 since with this deployment in the described scenario a normal service restart is also impossible even if the designated `Restart` button is pressed.

**Reply by AmirHosseinKarimi** [reply]
@thereis Changing the environments needs redeploy.

**Comment by Nexulo** [comment]
In my case, yes, absolutely. And how about you @thereis ?

**Reply by thereis** [reply]
I do like to have both scenarios, docker-compose services and dockerfile services, where I just simply click on restart and the new env vars will be on the container, rather than rebuilding it. The new env vars are pretty much runtime vars rather than build time.

**Reply by Nexulo** [reply]
It's exactly the same for me. It would be really helpful and save a lot of time.

**Comment by mredodos** [comment]
+1

**Comment by skusez** [comment]
> It would be cool if we could have a feature where we can simply restart the container in the server rather than triggering a new build of it.

This would save me so much time when debugging reployments

**Comment by jrvidotti** [comment]
+1 for this one!

**Comment by miguelfrmn** [comment]
+1 too

**Comment by mcanti-ntt** [comment]
+1
